services/migration_service/migration_files/20230118020300_drop_partial_indexes.sql

```diff
@@ -0,0 +1,108 @@
+-- +goose NO TRANSACTION
+-- +goose Up
+
+-- Drop partial str_ids indexes created with 20211202100726_add_str_id_indices.sql
+-- and recreate them without the constraining WHERE clause.
+-- This is being done as the psql query planner is not using these indexes many times.
+-- To avoid perf downtime we first create the new indexes and then drop the old ones.
+CREATE INDEX CONCURRENTLY IF NOT EXISTS runs_v3_idx_str_ids_primary_key_v2
+  ON runs_v3 (flow_id, run_id);
+
+DROP INDEX CONCURRENTLY IF EXISTS runs_v3_idx_str_ids_primary_key;
+
+CREATE INDEX CONCURRENTLY IF NOT EXISTS steps_v3_idx_str_ids_primary_key_v2
+  ON steps_v3 (flow_id, run_id, step_name);
+
+DROP INDEX CONCURRENTLY IF EXISTS steps_v3_idx_str_ids_primary_key;
+
+CREATE INDEX CONCURRENTLY IF NOT EXISTS tasks_v3_idx_flow_id_run_id_step_name_task_name
```

Comment:
This index has the same name as the one with the `WHERE ...` clauses. At best it doesn't create an index because of the `IF NOT EXISTS`; at worst it creates the index only for the next `DROP INDEX` to destroy it.
```suggestion
CREATE INDEX CONCURRENTLY IF NOT EXISTS tasks_v3_idx_flow_id_run_id_step_name_task_name_v2
```

Reply:
This is a very good catch, yes it's a bug, and your suggestion is correct. I don't know how I missed this!